core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java

```diff
@@ -189,44 +195,93 @@ public static Set<Pattern> getGlobalReplace() {
 		return globalReplace;
 	}
 
-	public static void cleanGlobalReplace() {
+	public static void clearGlobalReplace() {
 		globalReplace = new HashSet<>();
 	}
 
-	public static void addToThreadLocalReplace(Set<Pattern> hideRegexSet) {
-		if (hideRegexSet == null || hideRegexSet.isEmpty()) return;
-
-		if (threadLocalReplace.get() == null)
-			createThreadLocalReplace();
+	/**
+	 * Replace all thread-local hideRegexes. Always clears the current stack, even if the
+	 * new collection is null or empty.
+	 * This method should be used to initialize the stack of hideregexes at the start of a new thread
+	 * when there might be hideregexes to be carried over from a calling thread.
+	 *
+	 * @see org.frankframework.stream.ThreadConnector
+	 *
+	 * @param hideRegexCollection Collection of new hideRegexes. Can be null or empty.
+	 */
+	public static void setThreadLocalReplace(@Nullable Collection<Pattern> hideRegexCollection) {
+		clearThreadLocalReplace();
+		if (hideRegexCollection == null || hideRegexCollection.isEmpty()) return;
 
-		threadLocalReplace.get().addAll(hideRegexSet);
+		Deque<Pattern> stack = getOrCreateThreadLocalReplace();
+		stack.addAll(hideRegexCollection);
 	}
 
 	/**
-	 * Add regex to hide locally, meaning for specific threads/adapters.
-	 * This used to be LogUtil.setThreadHideRegex(String hideRegex)
+	 * Collection of regex strings to hide locally, meaning for specific threads/adapters.
+	 * Can return null when not used/initialized!
 	 */
-	public static void addToThreadLocalReplace(Pattern regex) {
-		if(regex == null) return;
+	@Nullable
+	public static Collection<Pattern> getThreadLocalReplace() {
+		Deque<Pattern> stack = threadLocalReplace.get();
+		if (stack == null) return null;
+		return Collections.unmodifiableCollection(stack);
+	}
```

Comment:
Is het nog handig dat dit synchronized is?

Reply:
Ik denk niet dat dit nodig is?